automation/check-patch.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash -xe
+# Patch/PR testing hook script
+#
+
+main() {
+    # We need to keep the VMs between runs for now because the containers are
+    # built directly on the VMs and building them all each time takes a long time
```

Comment:
I don't build the containers on the VM image. I use a separate storage volume for the containers that is persistent even after destroying the VM using 'vagrant destroy'. This saves me a lot of time during development since docker build takes so long the first time. 

see the DOCKER_CACHE_DOCKER option.  You'll have to have vagrant-cachier installed for it to work.

Reply:
That is a good hint. It is called VAGRANT_CACHE_DOCKER. vagrant-cachier is not needed for it. It just attaches an extra disk, but you can also set VAGRANT_CACHE_RPM. This one will need vagrant-cachier, and it speeds up RPM installations in the vagrant VM.

Reply:
@rmohr ha. I meant VAGRANT_CACHE_DOCKER :D 

Actually vagrant-cachier is needed for the docker container caching.  For some reason, vagrant won't let the separate libvirt volume used for the docker images persist between "vagrant destroy && vagrant up" without vagrant-cachier installed.

Reply:
Hmm I guess I can try VAGRANT_CACHE_DOCKER.
I already have VAGRANT_CACHE_RPM enabled, but its probably not doing much because I disabled NFS.